app/assets/javascripts/components/projects/ProjectBoard.js

```diff
@@ -8,39 +9,112 @@ import Sprints from "../stories/Sprints";
 import ProjectLoading from './ProjectLoading';
 
-export class ProjectBoard extends React.Component {
-  componentWillMount() {
-    this.props.fetchProjectBoard(this.props.projectId);
-  }
+const ProjectBoard = ({
+  projectId,
+  createStory,
+  closeHistory,
+  notifications,
+  removeNotification,
+  history,
+  projectBoard,
+  fetchProjectBoard,
+  dragDropStory,
+  chillyBinStories,
+  backlogSprints,
+  doneSprints,
+  fetchPastStories
+}) => {
+  const [newChillyBinStories, setNewChillyBinStories] = useState();
+  const [newBacklogSprints, setNewBacklogSprints] = useState();
+
+  useEffect(() => {
+    setNewBacklogSprints(backlogSprints);
+  }, [backlogSprints]);
+
+  useEffect(() => {
+    setNewChillyBinStories(chillyBinStories);
+  }, [chillyBinStories]);
+
+  useEffect(() => {
+    fetchProjectBoard(projectId);
+  }, [fetchProjectBoard, projectId]);
+
+  const onDragEnd = ({ source, destination, draggableId }) => {
+    const { sprintIndex: sprintDropIndex, columnId: dropColumn } = JSON.parse(destination.droppableId);
+    const { sprintIndex: sprintDragIndex, columnId: dragColumn } = JSON.parse(source.droppableId);
+    const { index: sourceIndex } = source;
+    const { index: destinationIndex} = destination;
+    const isSameColumn = dragColumn === dropColumn;
+    const destinationArray = getArray(dropColumn, newBacklogSprints, newChillyBinStories, sprintDropIndex); // stories of destination column
+    const sourceArray = getArray(dragColumn, newBacklogSprints, newChillyBinStories, sprintDragIndex); // stories of source column
+    const dragStory = sourceArray[sourceIndex];
+
+    if (!destination) {
+      return;
+    }
+
+    if (isSameColumn && sourceIndex === destinationIndex) {
+      return;
+    }
+
+    const newPosition = getNewPosition(
+      destinationIndex,
+      sourceIndex,
+      destinationArray,
+      isSameColumn,
+      dragStory.storyType,
+    );
+
+    const newStories = moveTask(
+      sourceArray,
+      destinationArray,
+      sourceIndex,
+      destinationIndex,
+    );
+
+    // Changing the column array order
+    if (dropColumn === 'chillyBin') {
+      setNewChillyBinStories(newStories);
+    }
 
-  render() {
-    if (!this.props.projectBoard.isFetched) {
-      return <ProjectLoading data-id="project-loading" />;
+    if (dropColumn === 'backlog') {
+      setNewBacklogSprints(getNewSprints(newStories, newBacklogSprints, sprintDropIndex));
     }
 
-    const { 
-      projectId, 
-      createStory, 
-      closeHistory,
-      notifications, 
-      removeNotification, 
-      history,
-      projectBoard
-    } = this.props;
-
-    return (
+    // Persisting the new array order
+    const newState = getNewState(isSameColumn, dropColumn, dragStory.state);
+    return dragDropStory(dragStory.id, dragStory.projectId, {
+      position: newPosition,
+      state: newState,
+    });
+  };
+
+  if (!projectBoard.isFetched) {
+    return <b>{I18n.t('loading')}</b>;
```

Comment:
I believe than here is 

`return <ProjectLoading data-id="project-loading" />;`

Instead of 

`return <b>{I18n.t('loading')}</b>;`